drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -86,8 +237,8 @@ impl FpgaPortMasks {
 /// the FPGAs share code, resulting in each one reporting in terms of ports
 /// 0-15.
 ///
-/// This is the FPGA -> logical mapping.
-const PORT_MAP: [PortLocation; 32] = [
+/// This is the logical -> physical mapping.
+const PORT_MAP: [PortLocation; NUM_PORTS as usize] = [
```

Comment:
Continuing the drive for type-safety, I wonder if we should enforces only indexing `PORT_MAP` with a `LogicalPort`, e.g. a new type:

```rust
struct PortMap([PortLocation; NUM_PORTS as usize]);

impl core::ops::Index<LogicalPort> for PortMap {
    type Output = PortLocation;

    fn index(&self, i: LogicalPort) -> &Self::Output {
        self.0[i.0]
    }
}

const PORT_MAP: PortMap = PortMap([
    // ...
]);
```

Reply:
We chatted a bit in DMs - I implemented `PortMap` and some supporting `impl`s for a Stronger Type.